Formula/ksync.rb

```diff
@@ -0,0 +1,36 @@
+class Ksync < Formula
+  desc "Sync files between your local system and a kubernetes cluster"
+  homepage "https://ksync.github.io/ksync/"
+  url "https://github.com/ksync/ksync.git",
+    :tag      => "0.4.5",
+    :revision => "9f40bf134329814a57e1a58d73b84761a2b06c73"
+  sha256 "bd47c5b5418bfc6d2f08abe855a52d4189f8bf380747cdd4e5c1a3ca0a87498a"
+  license "Apache-2.0"
+  depends_on "go" => :build
+
+  def install
+    # Set some stamps and build local CLI binary
+    time = `date +%Y%m%dT%H%M%S`.chomp
+    goversion = "go#{Formula["go"].version}"
+    commit = Utils.safe_popen_read("git", "rev-parse", "--short", "HEAD").chomp
+    system "go", "build", "-ldflags",
+              "-w -X github.com/ksync/ksync/pkg/ksync.GitCommit=#{commit} \
+              -X github.com/ksync/ksync/pkg/ksync.GitTag=#{version} \
+              -X github.com/ksync/ksync/pkg/ksync.BuildDate=#{time} \
+              -X github.com/ksync/ksync/pkg/ksync.VersionString=Homebrew \
+              -X github.com/ksync/ksync/pkg/ksync.GoVersion=#{goversion}",
+              *std_go_args,
+              "github.com/ksync/ksync/cmd/ksync"
+  end
+
+  # PLACEHOLDER: Need to find a way of running cleanup before uninstall
```

Comment:
Why is that needed?

Reply:
Self reminder. I can find no formula syntax for pre-uninstall cleanup steps.

Reply:
That's because there isn't any. What would be the result if this isn't ran?

Reply:
Two things of importance:
- Local config and binaries are not cleaned up
- Remote components (cluster side) are not removed

Reply:
The local files will be removed by brew because we're managing the package.

As for the remote files, I think you should just include that step in your documentation.

Reply:
Would brew remove files it doesn't know about? (the `init` command creates said files)

Reply:
Maybe adding a warning to the install would be good enough for the moment

Reply:
Brew would not remove files it doesn't know about. But it's up to users to learn how the tool works. Unless there's something that the user needs to know only for homebrew you shouldn't print a warning though.

It sounds like you want to clean up a staging area, and that's up to users.